pilot/pkg/model/sidecar.go

```diff
@@ -723,3 +723,13 @@ func needsPortMatch(ilw *IstioEgressListenerWrapper) bool {
 	return ilw.IstioListener != nil && ilw.IstioListener.Port.GetNumber() != 0 &&
 		protocol.Parse(ilw.IstioListener.Port.Protocol) != protocol.HTTP_PROXY
 }
+
+// newConfigHostsSetting clone vs config based on origin, then setting new vs config's hosts.
+func newVSConfigHostsSetting(origin config.Config, hosts string) config.Config {
+	newVSConfig := origin.DeepCopy()
+	newVirtualService := newVSConfig.Spec.(*networking.VirtualService)
+	newVirtualService.Hosts = []string{hosts}
+	newVSConfig.Spec = config.Spec(newVirtualService)
```

Comment:
redudant

Reply:
yes, this will be removed, no VS clone, but check duplicated when adding VS into slice